x-pack/legacy/plugins/ml/server/routes/data_frame_analytics.js

```diff
@@ -4,171 +4,302 @@
+  );
+
+  async function dfAnalyticsEvaluateHandler(context, request, response) {
+    try {
+      const results = await context.ml.mlClient.callAsCurrentUser('ml.evaluateDataFrameAnalytics', {
+        body: request.body,
+      });
+      return response.ok({
+        body: { ...results },
+      });
+    } catch (e) {
+      // Case: default
+      return response.internalError({ body: e.message });
+    }
+  }
+
+  router.post(
+    {
+      path: '/api/ml/data_frame/_evaluate',
+      validate: {
+        body: schema.object({ ...dataAnalyticsEvaluateSchema }),
+      },
     },
-  });
+    licensePreRoutingFactory(xpackMainPlugin, dfAnalyticsEvaluateHandler)
+  );
 
-  route({
-    method: 'DELETE',
-    path: '/api/ml/data_frame/analytics/{analyticsId}',
-    handler(request) {
-      const callWithRequest = callWithRequestFactory(elasticsearchPlugin, request);
-      const { analyticsId } = request.params;
-      return callWithRequest('ml.deleteDataFrameAnalytics', { analyticsId }).catch(resp =>
-        wrapError(resp)
+  async function dfAnalyticsExplainHandler(context, request, response) {
+    try {
+      const results = await context.ml.mlClient.callAsCurrentUser(
+        'ml.estimateDataFrameAnalyticsMemoryUsage',
+        {
+          body: request.body,
+        }
       );
+      return response.ok({
+        body: { ...results },
+      });
+    } catch (e) {
+      // Case: default
+      return response.internalError({ body: e.message });
+    }
+  }
+
+  router.post(
+    {
+      path: '/api/ml/data_frame/analytics/_explain',
+      validate: {
+        body: schema.object({ ...dataAnalyticsExplainSchema }),
+      },
     },
-    config: {
-      ...commonRouteConfig,
-    },
-  });
+    licensePreRoutingFactory(xpackMainPlugin, dfAnalyticsExplainHandler)
+  );
 
-  route({
-    method: 'POST',
-    path: '/api/ml/data_frame/analytics/{analyticsId}/_start',
-    handler(request) {
-      const callWithRequest = callWithRequestFactory(elasticsearchPlugin, request);
-      const options = {
-        analyticsId: request.params.analyticsId,
-      };
+  async function dfAnalyticsDeleteJobHandler(context, request, response) {
+    try {
+      const { analyticsId } = request.params;
+      const results = await context.ml.mlClient.callAsCurrentUser('ml.deleteDataFrameAnalytics', {
+        analyticsId,
+      });
+      return response.ok({
+        body: { ...results },
+      });
+    } catch (e) {
+      // Case: default
+      return response.internalError({ body: e.message });
+    }
+  }
 
-      return callWithRequest('ml.startDataFrameAnalytics', options).catch(resp => wrapError(resp));
+  router.delete(
+    {
+      path: '/api/ml/data_frame/analytics/{analyticsId}',
+      validate: {
+        params: schema.object({
+          analyticsId: schema.string(),
+        }),
+      },
     },
-    config: {
-      ...commonRouteConfig,
+    licensePreRoutingFactory(xpackMainPlugin, dfAnalyticsDeleteJobHandler)
```

Comment:
NIT: subjective opinion, but you could inline the handlers as they are all used only once. This will also helps when converting to TS as the `context, request, response` types will be inferred.

```typescript
licensePreRoutingFactory(xpackMainPlugin, (context, request, response) {
    try {
      const { analyticsId } = request.params;
      const results = await context.ml.mlClient.callAsCurrentUser('ml.deleteDataFrameAnalytics', {
        analyticsId,
      });
      return response.ok({
        body: { ...results },
      });
    } catch (e) {
      // Case: default
      return response.internalError({ body: e.message });
    }
  })
```

Reply:
Great suggestion! Changed in 1be4d620155fc0186cd7fedcaebb6c7ba5a51c9c